src/auth.ts

```diff
@@ -0,0 +1,31 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as core from '@actions/core';
+import * as io from '@actions/io';
+
+export async function configAuthentication(username: string, password: string) {
+  const directory: string = path.join(os.homedir(), '.m2');
+  await io.mkdirP(directory);
+  await write(directory, generate(username, password));
+}
+
+// only exported for testing purposes
+export function generate(
+  username = '${actions.username}',
+  password = '${actions.password}'
```

Comment:
I did this to allow for the option 2 as a possible fallback but would need to make changes to the `setup-java` such that undefined parameters are passed.  If this route looks reasonable I can make the changes and users could pass in CL options like I described in option 2.  If its extra noise I'm happy to remove it.

Reply:
Removed this so there isn't confusion.